components/brave_today/browser/brave_news_controller.cc

```diff
@@ -121,6 +130,34 @@ BraveNewsController::MakeRemote() {
   return remote;
 }
 
+void BraveNewsController::GetLocale(GetLocaleCallback callback) {
+  publishers_controller_.GetOrFetchPublishers(base::BindOnce(
+      [](GetLocaleCallback callback, Publishers publishers) {
+        base::flat_set<std::string> available_locales;
+        for (const auto& it : publishers) {
+          for (const auto& locale : it.second->locales)
+            available_locales.insert(locale);
+        }
+
+        // Locale is lang-COUNTRY but Brave News wants the format to be
+        // lang_COUNTRY.
+        const std::string locale =
+            brave_l10n::LocaleHelper::GetInstance()->GetLocale();
+        const std::string language_code = brave_l10n::GetLanguageCode(locale);
+        const std::string country_code = brave_l10n::GetCountryCode(locale);
+        const std::string brave_news_locale =
+            language_code + "_" + country_code;
+
+        if (base::Contains(available_locales, brave_news_locale))
+          return std::move(callback).Run(brave_news_locale);
+
+        // Fallback to en_US, if we can't match anything else.
+        // TODO(fallaciousreasoning): Implement more complicated fallback
+        std::move(callback).Run("en_US");
```

Comment:
nit: since this will only change when sources is re-fetched, perhaps we should store this on `PublishersController`?

Reply:
Done